src/main/java/com/shootdoori/match/service/TeamMemberService.java

```diff
@@ -185,16 +182,43 @@ public TeamMemberResponseDto delegateLeadership(Long teamId, Long currentUserId,
 
     public TeamMemberResponseDto delegateViceLeadership(Long teamId, Long currentUserId,
         Long targetMemberId) {
-        // TODO: 부회장 위임 로직 구현 필요 (권한/자기위임/같은 팀 검증 후 역할 변경)
-        throw new NoPermissionException();
+        List<TeamMember> members = prepareDelegationMembers(teamId, currentUserId, targetMemberId);
+
+        TeamMember currentMember = members.get(0);
+        TeamMember targetMember = members.get(1);
+
+        currentMember.delegateViceLeadership(targetMember);
+
+        return teamMemberMapper.toTeamMemberResponseDto(targetMember);
```

Comment:
```suggestion
        TeamMember delegatedMember = currentMember(teamId, currentUserId)
                .delegateLeadership(targetMember(targetMemberId));
        return teamMemberMapper.toTeamMemberResponseDto(delegatedMember);
```

- 어떤 형태기 직관적일지, 재사용 가능한지 계속 고민해보셔요